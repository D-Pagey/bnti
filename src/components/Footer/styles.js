import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { margin, padding } from 'polished';
import { colour, fontSize, mediaQuery, spacing } from '../../tokens';

export const Wrapper = styled.div`
    align-items: center;
    color: ${colour.white};
    display: flex;
    flex-direction: column;
    ${padding(spacing.large, spacing.small)};
    ${margin(0, 'auto')};
    max-width: 650px;

    @media ${mediaQuery.min.medium} {
        display: grid;
        grid-template-columns: 50% 50%;
        grid-template-rows: repeat(4, 25%);
        justify-items: center;
    }
`;

export const Title = styled.h3`
    font-size: ${fontSize.huge};
    text-align: center;
    ${margin(0)}
`;

export const White = styled.span`
    color: ${colour.white};
`;

export const Blue = styled.span`
    color: ${colour.blue};
`;

export const List = styled.ul`
    border-bottom: 1px solid ${colour.white};
    display: grid;
    grid-template-columns: 50% 50%;
    grid-row-gap: 5px;
    list-style: none;
    ${margin(spacing.large, 0, spacing.tiny)};
    ${padding(0, 0, spacing.large)};
    width: 265px;

    @media ${mediaQuery.min.medium} {
        border-top: 1px solid ${colour.white};
        grid-column: 2 / 3;
        grid-row: 1 / 5;
        ${margin(0)};
        ${padding(spacing.large, 0)};
    }
`;

export const ListItem = styled.li`
    text-align: center;
`;

export const FooterLink = styled(Link)`
    color: ${colour.white};
    text-decoration: none;
`;

export const LinkLegal = styled.p`
    ${margin(spacing.small, 0)};
`;

export const SocialWrapper = styled.div`
    display: flex;
`;

export const Logo = styled.img`
    ${margin(spacing.small, spacing.tiny)}
`;
