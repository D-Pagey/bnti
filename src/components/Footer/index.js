import React from 'react';
import facebook from './assets/facebook.png';
import twitch from './assets/twitch.png';
import twitter from './assets/twitter.png';
import youtube from './assets/youtube.png';
import * as S from './styles';

const Footer = () => (
    <S.Wrapper>
        <S.Title>
            <S.White>BR</S.White>
            <S.Blue>Battles</S.Blue>
        </S.Title>
        <S.List>
            <S.ListItem>
                <S.FooterLink to="/">Home</S.FooterLink>
            </S.ListItem>
            <S.ListItem>
                <S.FooterLink to="/about">About</S.FooterLink>
            </S.ListItem>
            <S.ListItem>
                <S.FooterLink to="/">FAQ</S.FooterLink>
            </S.ListItem>
            <S.ListItem>
                <S.FooterLink to="/jobs">Jobs</S.FooterLink>
            </S.ListItem>
            <S.ListItem>
                <S.FooterLink to="/">Tournaments</S.FooterLink>
            </S.ListItem>
            <S.ListItem>
                <S.FooterLink to="/">Stats</S.FooterLink>
            </S.ListItem>
            <S.ListItem>
                <S.FooterLink to="/">Account</S.FooterLink>
            </S.ListItem>
            <S.ListItem>
                <S.FooterLink to="/signin">Sign In</S.FooterLink>
            </S.ListItem>
        </S.List>
        <S.LinkLegal>
            <S.FooterLink to="/">Privacy Policy</S.FooterLink>
        </S.LinkLegal>
        <S.LinkLegal>
            <S.FooterLink to="/">Terms of Use</S.FooterLink>
        </S.LinkLegal>
        <S.SocialWrapper>
            <S.Logo src={twitter} alt="twitter" />
            <S.Logo src={facebook} alt="facebook" />
            <S.Logo src={youtube} alt="youtube" />
            <S.Logo src={twitch} alt="twitch" />
        </S.SocialWrapper>
    </S.Wrapper>
);

export default Footer;
