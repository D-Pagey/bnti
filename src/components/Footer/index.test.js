import React from 'react';
import Footer from '.';

describe('Footer component', () => {
    it('should render', () => {
        const { container } = renderWithRouter(<Footer />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
