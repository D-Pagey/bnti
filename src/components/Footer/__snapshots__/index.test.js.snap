// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should render 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 1.5rem;
  padding-right: 0.75rem;
  padding-bottom: 1.5rem;
  padding-left: 0.75rem;
  margin-top: 0;
  margin-right: auto;
  margin-bottom: 0;
  margin-left: auto;
  max-width: 650px;
}

.c1 {
  font-size: 2.5rem;
  text-align: center;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
}

.c2 {
  color: #FFF;
}

.c3 {
  color: #00A7CB;
}

.c4 {
  border-bottom: 1px solid #FFF;
  display: grid;
  grid-template-columns: 50% 50%;
  grid-row-gap: 5px;
  list-style: none;
  margin-top: 1.5rem;
  margin-right: 0;
  margin-bottom: 0.5rem;
  margin-left: 0;
  padding-top: 0;
  padding-right: 0;
  padding-bottom: 1.5rem;
  padding-left: 0;
  width: 265px;
}

.c5 {
  text-align: center;
}

.c6 {
  color: #FFF;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7 {
  margin-top: 0.75rem;
  margin-right: 0;
  margin-bottom: 0.75rem;
  margin-left: 0;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9 {
  margin-top: 0.75rem;
  margin-right: 0.5rem;
  margin-bottom: 0.75rem;
  margin-left: 0.5rem;
}

@media (min-width:768px) {
  .c0 {
    display: grid;
    grid-template-columns: 50% 50%;
    grid-template-rows: repeat(4,25%);
    justify-items: center;
  }
}

@media (min-width:768px) {
  .c4 {
    border-top: 1px solid #FFF;
    grid-column: 2 / 3;
    grid-row: 1 / 5;
    margin-top: 0;
    margin-right: 0;
    margin-bottom: 0;
    margin-left: 0;
    padding-top: 1.5rem;
    padding-right: 0;
    padding-bottom: 1.5rem;
    padding-left: 0;
  }
}

<div
  class="c0"
>
  <h3
    class="c1"
  >
    <span
      class="c2"
    >
      BR
    </span>
    <span
      class="c3"
    >
      Battles
    </span>
  </h3>
  <ul
    class="c4"
  >
    <li
      class="c5"
    >
      <a
        class="c6"
        href="/"
      >
        Home
      </a>
    </li>
    <li
      class="c5"
    >
      <a
        class="c6"
        href="/about"
      >
        About
      </a>
    </li>
    <li
      class="c5"
    >
      <a
        class="c6"
        href="/"
      >
        FAQ
      </a>
    </li>
    <li
      class="c5"
    >
      <a
        class="c6"
        href="/jobs"
      >
        Jobs
      </a>
    </li>
    <li
      class="c5"
    >
      <a
        class="c6"
        href="/"
      >
        Tournaments
      </a>
    </li>
    <li
      class="c5"
    >
      <a
        class="c6"
        href="/"
      >
        Stats
      </a>
    </li>
    <li
      class="c5"
    >
      <a
        class="c6"
        href="/"
      >
        Account
      </a>
    </li>
    <li
      class="c5"
    >
      <a
        class="c6"
        href="/signin"
      >
        Sign In
      </a>
    </li>
  </ul>
  <p
    class="c7"
  >
    <a
      class="c6"
      href="/"
    >
      Privacy Policy
    </a>
  </p>
  <p
    class="c7"
  >
    <a
      class="c6"
      href="/"
    >
      Terms of Use
    </a>
  </p>
  <div
    class="c8"
  >
    <img
      alt="twitter"
      class="c9"
      src="twitter.png"
    />
    <img
      alt="facebook"
      class="c9"
      src="facebook.png"
    />
    <img
      alt="youtube"
      class="c9"
      src="youtube.png"
    />
    <img
      alt="twitch"
      class="c9"
      src="twitch.png"
    />
  </div>
</div>
`;
